### ISSUE TITLE
Footnote numbers can get split across lines

### ISSUE THREAD

**Issue by mojavelinux** [asciidoctor/asciidoctor-pdf#2297]
If a footnote number falls the end the line, the brackets and numbers can get split across lines. For example:

```
Lorem ipsum dolor sit amet, consectetur adipiscing elit.[
1]
```

This happens because the footnote number is three separate fragments and Prawn has no way to correlate them.

Furthermore, if the footnote number is the last text in the paragraph, it will be placed too high on the line, making it overlap the preceding line.

**Comment by mojavelinux** [comment]
One way to solve this problem is to put the square brackets around the number inside the link. This appearance is consistent with how footnotes are done on Wikipedia and thus a reasonable style to use as a default.

The other solution is to implement wrap avoidance (i.e., nowrap) in the arranger.

**Comment by mojavelinux** [comment]
Since we're going to proceed with a style change, I'm scheduling this fix for the 2.3.x release line.

**Comment by mojavelinux** [comment]
I'm going to apply the style change in a separate issue that will be schedule for 3.0.0.
